misc/scripts/build.sh

```diff
@@ -183,90 +254,11 @@ function prompt_optdepends() {
-                    if [[ $i != "n" && $i != "y" ]] && ((i > ${#suggested_optdeps[@]})); then
-                        local skip_opt+=("$i")
-                        unset 'choices[$choice_inc]'
-                    fi
-                    { ignore_stack=true; ((choice_inc++)); }
-                done
-                if [[ -n ${skip_opt[*]} ]]; then
-                    fancy_message warn "${BGreen}${skip_opt[*]}${NC} has exceeded the maximum number of optional dependencies. Skipping"
-                fi
-
-                # Did we get actual answers?
-                if [[ ${choices[0]} != "n" && ${choices[0]} != "0" ]]; then
-                    for i in "${choices[@]}"; do
-                        # Set our user array that started at 1 down to 0 based
-                        local not_installed_yet_optdeps+=("${suggested_optdeps[$((i - 1))]}")
-                    done
-                    if [[ -n ${not_installed_yet_optdeps[*]} ]]; then
-                        fancy_message info "Selecting packages ${BCyan}${not_installed_yet_optdeps[*]%%:\ *}${NC}"
-                        local log_depends log_depends_str
-                        dep_const.format_control optdepends log_depends
-                        dep_const.comma_array log_depends log_depends_str
-                        deblog "Suggests" "${log_depends_str}"
-                        fancy_message info "Installing selected optional dependencies"
-                    fi
-                else # Did we get 0 or n?
-                    # Add everything to Suggests
-                    local log_depends log_depends_str
-                    dep_const.format_control optdepends log_depends
-                    dep_const.comma_array log_depends log_depends_str
-                    deblog "Suggests" "${log_depends_str}"
-                fi
-            else # If `-B` is being used
-                # We can log everything from optdepends to Suggests
-                # shellcheck disable=SC2034
-                local log_depends log_depends_str
-                dep_const.format_control optdepends log_depends
-                dep_const.comma_array log_depends log_depends_str
-                deblog "Suggests" "${log_depends_str}"
-            fi
-        fi
+        prompt_optdepends
     fi
-
-    # shellcheck disable=SC2034
-    local depends_for_logging out_str
     if [[ -n ${pacdeps[*]} ]]; then
         for i in "${pacdeps[@]}"; do
-            (
-                #shellcheck disable=SC1090
-                source "$METADIR/$i"
-                if [[ -n $_gives ]]; then
-                    echo "$_gives" | tee -a "${PACDIR}-gives-${pacname}" > /dev/null
-                else
-                    echo "$_name" | tee -a "${PACDIR}-gives-${pacname}" > /dev/null
-                fi
-            )
+            awk -F'=' '/^_gives=/{gives=$2} /^_name=/{name=$2} END{print (gives ? gives : name)}' "${METADIR}/${i}" >> "${PACDIR}-gives-${pacname}"
```

Comment:
If this is actually easier then fine, but I'm just saying I don't like the amount of awk creeping into the codebase.